src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,113 @@
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        String query = "SELECT product_id, name, price FROM products "
+                + "WHERE product_id = ? AND deleted = 0";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement select = connection.prepareStatement(query);
+            select.setLong(1, id);
+            ResultSet productRow = select.executeQuery();
+            if (productRow.next()) {
+                Long productId = productRow.getLong("product_id");
+                String name = productRow.getString("name");
+                BigDecimal price = productRow.getBigDecimal("price");
+                Product product = new Product(name, price);
+                product.setId(productId);
+                return Optional.of(product);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get the product", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Product update(Product item) {
+        String query = "UPDATE products SET name = ?, price = ? "
+                + "WHERE product_id = ? AND deleted = 0";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement update = connection.prepareStatement(query);
+            update.setString(1, item.getName());
+            update.setBigDecimal(2, item.getPrice());
+            update.setLong(3, item.getId());
+            update.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update product", e);
+        }
+        return item;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "UPDATE products SET deleted = 1 WHERE product_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement delete = connection.prepareStatement(query);
+            delete.setLong(1, id);
+            int deletedRow = delete.executeUpdate();
+            if (deletedRow == 1) {
+                return true;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update product", e);
+        }
+        return false;
+    }
+
+    @Override
+    public List<Product> getAll() {
+        String query = "SELECT product_id, name, price FROM products WHERE deleted = 0";
+        List<Product> products = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement select = connection.prepareStatement(query);
+            ResultSet productRow = select.executeQuery();
+            while (productRow.next()) {
+                Long productId = productRow.getLong("product_id");
+                String name = productRow.getString("name");
+                BigDecimal price = productRow.getBigDecimal("price");
+                Product product = new Product(name, price);
```

Comment:
You used the same code in method get(). You had better create a separate private method

Reply:
ok, thank you)